### PR TITLE
Test for expected exception instead of xfail

### LINKED ISSUE #1109: Test for expected exception
Make use of pytest.raises instead of using pytest.mark.xfail

https://github.com/pymc-labs/pymc-marketing/blob/8a284dc65515e2900d7b63d0d94be8f4a02611b2/tests/test_model_builder.py#L260-L265

### PR DESCRIPTION
**Pull request by nhan295**
## Related Issue

- [x] Closes #1109
- [ ] Related to #

## Checklist

- [ ] Checked that [the pre-commit linting/style checks pass](https://docs.pymc.io/en/latest/contributing/python_style.html)
- [ ] Included tests that prove the fix is effective or that the new feature works
- [ ] Added necessary documentation (docstrings and/or example notebooks)
- [ ] If you are a pro: each commit corresponds to a [relevant logical change](https://wiki.openstack.org/wiki/GitCommitMessages#Structural_split_of_changes)

## Modules affected

- [ ] MMM
- [ ] CLV

## Type of change

- [ ] New feature / enhancement
- [ ] Bug fix
- [ ] Documentation
- [ ] Maintenance
- [ ] Other (please specify):

----
📚 Documentation preview 📚: https://pymc-marketing--1114.org.readthedocs.build/en/1114/

### CODE REVIEW

File: tests/test_model_builder.py [hunk 1]
```diff
@@ -257,12 +257,14 @@ def test_fit_no_t(toy_X):
...
-def test_fit_dup_Y(toy_X):
-    # create redundant target column in X
+def test_fit_dup_Y(toy_X, toy_y):
+    # Create redundant target column in X
     toy_X = pd.concat((toy_X, toy_y), axis=1)
     model_builder = ModelBuilderTest()
-    model_builder.fit(X=toy_X, chains=1, draws=100, tune=100)
+
+    # Expecting an exception due to redundant target columns
+    with pytest.raises(ValueError, match="Redundant target column found in input data."):
```
Comment (juanitorduz) [inline tests/test_model_builder.py:266]: I think we need to change the `match` statement, as the test are failing:

```python
    def test_fit_dup_Y(toy_X, toy_y):
        # Create redundant target column in X
        toy_X = pd.concat((toy_X, toy_y), axis=1)
        model_builder = ModelBuilderTest()
    
        # Expecting an exception due to redundant target columns
>       with pytest.raises(ValueError, match="Redundant target column found in input data."):
E       AssertionError: Regex pattern did not match.
E        Regex: 'Redundant target column found in input data.'
E        Input: "X includes a column named 'output', which conflicts with the target variable."
```

**Review by juanitorduz — CHANGES_REQUESTED** [review]

**Comment by juanitorduz** [comment]
Thank you @nhan295 ! I think is *almost* there.

File: tests/test_model_builder.py (hunk 1 above)
Comment (nhan295) [reply tests/test_model_builder.py:266]: ok let me check

**Comment by wd60622** [comment]
pre-commit.ci autofix

File: tests/test_model_builder.py [hunk 2]
```diff
@@ -257,12 +257,14 @@ def test_fit_no_t(toy_X):
...
-    # create redundant target column in X
+def test_fit_dup_Y(toy_X, toy_y):
     toy_X = pd.concat((toy_X, toy_y), axis=1)
     model_builder = ModelBuilderTest()
-    model_builder.fit(X=toy_X, chains=1, draws=100, tune=100)
+
+    with pytest.raises(ValueError) as excinfo:
+        model_builder.fit(X=toy_X, chains=1, draws=100, tune=100)
+
+    assert "Redundant target column found in input data." in str(excinfo.value)
```
Comment (wd60622) [inline tests/test_model_builder.py:267]: This error message is incorrect

**Review by wd60622 — CHANGES_REQUESTED** [review]

**Comment by wd60622** [comment]
pre-commit.ci autofix

**Comment by wd60622** [comment]
FYI @nhan295 run `pre-commit install` locally to get rid the pre-commit failures

**Review by juanitorduz — APPROVED** [review]

**Comment by juanitorduz** [comment]
Thank you @nhan295 !

**Pull request merged**